app/services/sign_in/assertion_validator.rb

```diff
@@ -25,14 +28,14 @@ def validate_service_account_config
       end
     end
 
-    def validate_iss
-      if decoded_assertion.iss != audience
+    def validate_issuer
+      if issuer != audience
         raise Errors::ServiceAccountAssertionAttributesError.new message: 'Assertion issuer is not valid'
       end
     end
 
     def validate_audience
-      unless decoded_assertion.aud.match(token_route)
+      unless Array(assertion_audience).include?(token_route)
```

Comment:
just a nitpick, but I'd move the `Array` part of this into the memoized function to be consistent with the `scopes` function that does this